### PR TITLE
Copy runtime pack resource assemblies on self-contained build/publish.

### LINKED ISSUE #3068: Support framework satellite assemblies in self-contained deployment
The runtime packs for Microsoft.WindowsDesktop.App now have satellites assemblies. However, they are not used by self-contained build or publish.

They should be deployed exactly as satellites from a NuGet package. That is:

1. If `$(SatelliteResourceLanguages)` is not specified, then all satellites are deployed.

2. If `$(SatelliteResourceLanguages)` is specified, then only matching satellites are deployed.

cc @dsplaisted @leecow @vivmishra

This is needed for .NET Core 3.0 Preview 6

### PR DESCRIPTION
**Pull request by peterhuene**
This commit implements copying resource assemblies for self-contained
build/publish from runtime packs.

Closes #3068.

### CODE REVIEW

File: src/Tasks/Microsoft.NET.Build.Tasks/ResolveRuntimePackAssets.cs [hunk 1]
```diff
@@ -99,9 +101,56 @@ void AddAsset(string assetPath, string assetType)
...
+        {
+            var culture = Path.GetFileName(cultureDirectory);
+
+            if (this.SatelliteResourceLanguages.Length > 1 &&
+                !this.SatelliteResourceLanguages.Any(lang => string.Equals(lang.ItemSpec, culture, StringComparison.OrdinalIgnoreCase)))
+            {
+                yield break;
+            }
+
+            foreach (var file in Directory.EnumerateFiles(cultureDirectory))
```
Comment (nguerrera) [inline src/Tasks/Microsoft.NET.Build.Tasks/ResolveRuntimePackAssets.cs]: Is it worth filtering to *.resources.dll here? We got burned by over-aggressively picking up random files fro runtime pack before. I realize the manifests should be the real solution here, but I wonder if we should more conservative in the meantime.

File: src/Tests/Microsoft.NET.TestFramework/ProjectConstruction/TestProject.cs [hunk 2]
```diff
@@ -426,35 +432,5 @@ private bool ReferenceAssembliesAreInstalled(string targetFrameworkVersion)
             var requestedReferenceAssembliesPath = Path.Combine(new DirectoryInfo(net461referenceAssemblies).Parent.FullName, targetFrameworkVersion);
             return Directory.Exists(requestedReferenceAssembliesPath);
         }
-
-        public override string ToString()
```
Comment (nguerrera) [inline src/Tests/Microsoft.NET.TestFramework/ProjectConstruction/TestProject.cs]: I assume somebody did this for debugging at some point? I think I agree it's not worth maintaining.

**Review by nguerrera — APPROVED** [review]

File: src/Tests/Microsoft.NET.TestFramework/ProjectConstruction/TestProject.cs (hunk 2 above)
Comment (peterhuene) [reply src/Tests/Microsoft.NET.TestFramework/ProjectConstruction/TestProject.cs]: That was my assumption; I looked high and low for any usages and couldn't find any.

File: src/Tasks/Microsoft.NET.Build.Tasks/ResolveRuntimePackAssets.cs (hunk 1 above)
Comment (peterhuene) [reply src/Tasks/Microsoft.NET.Build.Tasks/ResolveRuntimePackAssets.cs]: I can add that.

**Pull request merged**